### PR TITLE
Create Separate Transformer for promo, price, new arrival and categor…

### PR DESCRIPTION
**Pull request by xayang**
…y associations

### CODE REVIEW

File: modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/package.scala [hunk 1]
```diff
@@ -1,5 +1,8 @@
 package com.hbc.streams.product
 
 package object transformer {
-  object instances extends TransformerInstances
```
Comment (josehbc) [inline modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/package.scala]: keep `TransformerInstances` and in the body import all of the below. That way you don't have to change topologies.

File: modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/PriceProdTransformerInstance.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        ProdsBySknName,
+        Duration.ofMinutes(2)
+      ) {
+        override def transform(
+          key: MultiBannerSkuKey,
+          value: PriceValue
+        ): KeyValue[MultiBannerProductKey, Product] = {
+          val now: Instant        = Instant.now
+          val priceEnd: Instant   = value.online_to.getOrElse(Instant.MAX)
+          val ClearancePriceCodes = Set("22", "23", "24")
```
Comment (josehbc) [inline modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/PriceProdTransformerInstance.scala]: move this outside of `transform`

File: modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/package.scala (hunk 1 above)
Comment (xayang) [reply modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/package.scala]: Do multiple traits inheritance to keep TransformerInstance. Then don't need to add cap for these separate transformers.

File: modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/PriceProdTransformerInstance.scala (hunk 2 above)
Comment (xayang) [reply modules/product-processor/src/main/scala/com/hbc/streams/product/transformer/PriceProdTransformerInstance.scala]: Move ClearancePriceCodes and assignment ids to config

**Pull request merged**